### ISSUE TITLE
Trace option

### ISSUE THREAD

**Issue by ganeshv** [pegjs/pegjs#121]
TLDR: Patch for tracing rules, helps greatly with debugging of new grammars.

While pegjs is way easier on the brain than BNF-based grammars, debugging a new grammar can be difficult. The generated code is so straightforward and simple that one can easily insert debug prints in rule-matching functions, but that gets wiped the moment you change a rule and regenerate the parser.

So I wrote a small patch which will print a trace of all the rules, the state of the input string at the point the rule is entered, and what was matched at leaving a rule.

To use it, you must invoke pegjs with the --trace command line option. e.g.  
`pegjs --export-var parser --track-line-and-column --trace foo.pegjs`

Additionally, you need to add a few options while invoking the parser:

```
var ast = parser.parse(source, {'startRule': 'start', 'trace': ['BlockList','Function',
'StatementBlock',  'IfStatement', 'EmptyStatement', 'EOS', 'Command' ,
'StringLiteral', 'Concat'], 'tabspace': 2, maxrules: source.length * 100});
```
-  The `trace` option is a list of rules for which trace output is emitted. This is to enable you to quickly narrow down the problem rule without wading through a ton of spam from very low-level tokenizer rules.
-  The `tabspace` option specifies how many spaces to indent each new rule.
-  The `maxrules` option is a half-assed safety net to catch infinite loops.

Sample output:

```
 Enter:BlockList:1:1:ls -l | gr
   Enter:Function:1:1:ls -l | gr
   Leave:Function:null
   Enter:StatementBlock:1:1:ls -l | gr
   ....
```

We entered BlockList, input line 1, column 1. Contents of input string (next 10 chars) "ls -l | gr"
BlockList = Function / StatementBlock. As you can see, the Function rule didn't match, so we try StatementBlock..

```
...
               Leave:SimpleCommand:{"ARGLIST":["ls","-l"]}
               Enter:SimpleCommand:1:9:grep foo
```

SimpleCommand matched and returned {"ARGLIST"..}. Now we try another SimpleCommand, input position is line 1, column 9, next few characters are "grep foo"...

Let me know if this sounds interesting. Code is lying in my fork, "trace" branch.
